### PR TITLE
Rpc refactor

### PR DESCRIPTION
**Pull request by keefertaylor**
Refactor RPCs to be pure model objects. Move all logic into ResponseAdapter. Add unit tests.

### CODE REVIEW

**Comment by keefertaylor** [comment]
I'm not really sure why Travis is kicking off four builds and 2 of them are unhappy, but only the latter two seem to matter...

File: Tests/RPCResponseHandlerTest.swift [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+}
+
+class RPCResponseHandlerTest: XCTestCase {
+  /// Response handler to test.
+  let responseHandler = RPCResponseHandler()
+
+  /// Test values.
+  let testURL = URL(string: "http://github.com/keefertaylor/TezosKit")!
+
+  let testErrorString = "flagrant error!"
```
Comment (jlaws) [inline Tests/RPCResponseHandlerTest.swift]: use
static let 
for constants

File: Tests/RPCResponseHandlerTest.swift [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+
+    guard let tezosKitError = error as? TezosClientError else {
+      XCTFail()
+      return
+    }
+
+    XCTAssertEqual(tezosKitError.kind, .unexpectedResponse)
+    XCTAssertEqual(tezosKitError.underlyingError, nil)
+  }
+
```
Comment (jlaws) [inline Tests/RPCResponseHandlerTest.swift]: // MARK: - Helpers

or something split the two groups

File: Tests/RPCResponseHandlerTest.swift (hunk 1 above)
Comment (keefertaylor) [reply Tests/RPCResponseHandlerTest.swift]: done

File: Tests/RPCResponseHandlerTest.swift (hunk 2 above)
Comment (keefertaylor) [reply Tests/RPCResponseHandlerTest.swift]: Done

**Pull request merged**